chia/server/start_service.py

```diff
@@ -238,23 +236,14 @@ async def setup_process_global_state(self) -> None:
 
         global main_pid
         main_pid = os.getpid()
-        if sys.platform == "win32" or sys.platform == "cygwin":
-            # pylint: disable=E1101
-            signal.signal(signal.SIGBREAK, self._accept_signal)
-            signal.signal(signal.SIGINT, self._accept_signal)
-            signal.signal(signal.SIGTERM, self._accept_signal)
-        else:
-            loop = asyncio.get_running_loop()
-            loop.add_signal_handler(
-                signal.SIGINT,
-                functools.partial(self._accept_signal, signal_number=signal.SIGINT),
-            )
-            loop.add_signal_handler(
-                signal.SIGTERM,
-                functools.partial(self._accept_signal, signal_number=signal.SIGTERM),
-            )
+        setup_sync_signal_handler(handler=self._accept_signal)
 
-    def _accept_signal(self, signal_number: int, stack_frame: Optional[FrameType] = None) -> None:
+    def _accept_signal(
+        self,
+        signal_number: int,
+        stack_frame: Optional[FrameType],
+        loop: asyncio.AbstractEventLoop,
+    ) -> None:
         self._log.info(f"got signal {signal_number}")
```

Comment:
I think this signal handler is problematic. It's kind of unrelated to your patch though. I think this might be the reason I sometimes have to `kill -9` left-over processes. if we won't tell the main process that the user wants to shut down, we should at least shut down *this* process, not just drop it.

Reply:
> Are you referring to having to kill workers? Or main processes.

I believe worker processes

>  Why would the subprocesses get signals to begin with? Does the signal to the main process get propagated to all? 

I think there are certain failure modes where we don't properly shut down the workers. Those worker processes can then not be killed with `SIGINT`, but only `SIGKILL`